Makefile

```diff
@@ -1,24 +1,73 @@
-.PHONY: all build-container build-tar build push-container push-tar push clean vet fmt version
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build the node-problem-detector image.
+
+.PHONY: all build-container build-tar build push-container push-tar push clean vet fmt version Dockerfile
 
 all: build
 
+# VERSION is the version of the binary.
 VERSION := $(shell git describe --tags --dirty)
 
+# TAG is the tag of the container image, default to binary version.
 TAG ?= $(VERSION)
 
+# PROJ is the image project.
+PROJ?=gcr.io/google_containers
+
+# UPLOAD_PATH is the cloud storage path to upload release tar.
 UPLOAD_PATH ?= gs://kubernetes-release
 # Trim the trailing '/' in the path
 UPLOAD_PATH := $(shell echo $(UPLOAD_PATH) | sed '$$s/\/*$$//')
 
-PROJ ?= google_containers
-
+# PKG is the package name of node problem detector repo.
 PKG := k8s.io/node-problem-detector
 
+# PKG_SOURCES are all the go source code.
 PKG_SOURCES := $(shell find pkg cmd -name '*.go')
 
+# TARBALL is the name of release tar. Include binary version by default.
 TARBALL := node-problem-detector-$(VERSION).tar.gz
 
-IMAGE := gcr.io/$(PROJ)/node-problem-detector:$(TAG)
+# IMAGE is the image name of the node problem detector container image.
+IMAGE := $(PROJ)/node-problem-detector:$(TAG)
+
+# ENABLE_JOURNALD enables build journald support or not. Building journald support needs libsystemd-dev
+# or libsystemd-journal-dev.
+# TODO(random-liu): Build NPD inside container.
+ENABLE_JOURNALD?=1
```

Comment:
Nit: spaces around the assignment operators missing to keep this in-sync with the other variables here.